src/DebugEngineHost/RegistryMonitor.cs

```diff
@@ -0,0 +1,148 @@
+﻿// // Copyright (c) Microsoft. All rights reserved.
+// // Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Win32;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Runtime.Remoting.Messaging;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using static System.Collections.Specialized.BitVector32;
+
+namespace Microsoft.DebugEngineHost
+{
+    internal class RegistryMonitor : IDisposable
+    {
+        #region Native Methods
+
+        /// <summary>
+        /// Filter for notifications reported by <see cref="RegistryMonitor"/>.
+        /// </summary>
+        [Flags]
+        public enum RegChangeNotifyFilter
+        {
+            /// <summary>Notify the caller if a subkey is added or deleted.</summary>
+            Key = 1,
```

Comment:
>Key [](http://example.com/codeflow?start=12&length=3)

nit: It is slightly weird that these names don't match up with MSDN. Might be worth at least including the Win32 constant names (ex: `REG_NOTIFY_CHANGE_NAME`): [https://learn.microsoft.com/en-us/windows/win32/api/winreg/nf-winreg-regnotifychangekeyvalue](https://learn.microsoft.com/en-us/windows/win32/api/winreg/nf-winreg-regnotifychangekeyvalue)